src/wms/src/test/java/org/geoserver/wms/wms_1_1_1/GetMapIntegrationTest.java

```diff
@@ -1988,4 +1988,27 @@ public void testLayoutLegendStyleTextFitBox() throws Exception {
         sInfo.setLegend(null);
         catalog.save(sInfo);
     }
+
+    @Test
+    public void testLayoutLegendWithSpecifiedTargetSize() throws Exception {
+        File layouts = getDataDirectory().findOrCreateDir("layouts");
+        URL layout = GetMapIntegrationTest.class.getResource("../test-layout-with-size.xml");
+        FileUtils.copyURLToFile(layout, new File(layouts, "test-layout-with-size.xml"));
+        BufferedImage image =
```

Comment:
So here you're asking for a map that's actually smaller than the legend decoration, so you expect the decoration to cover all of it, map not even being visible, and the decoration be cut as it does not fit.
Please clarify it in a comment.